### PR TITLE
Rework article list as class-based view

### LINKED ISSUE #1134: URLS for filtered articles
Generate URLs for filtered articles so we can directly link to pre-faceted article lists.

Development Work:
- [ ] Swap filter variables from session to GET parameters

### LINKED ISSUE #3199: Articles occasionally duplicate between front end 'articles' listing pages
**Describe the bug**
When articles are viewed on the front end of Janeway and the user clicks between pages (e.g. 1, 2, 3), the articles that are listed occasionally duplicate. E.g. https://lifewritingannual.openlibhums.org/articles/?page=1 'Editor's Farewell' appears on p1, and again on p.2: https://lifewritingannual.openlibhums.org/articles/?page=2

**Janeway version**
1.4.2.

**Expected behavior**
Articles should not duplicate between page lists

**Desktop (please complete the following information):**
 - macOS Monterey

### LINKED ISSUE #3425: Migrate Materialize Django helper
We'd like to move from our fork of 2016 [Django Materialize](https://github.com/BirkbeckCTP/django-materialize) to [django-materializecss-form](https://github.com/kalwalkden/django-materializecss-form).

This also involves upgrading from v0.100.2.

## Upgrade Materialize
- [x] Upgrade Materialize to v1.2.2
- [x] Handle breaking changes between v0.100.2 and v1.0.0 https://github.com/Dogfalo/materialize/blob/v1-dev/v1-upgrade-guide.md
- [x] Handle breaking changes between v1.0.0 and v1.2.2 https://github.com/materializecss/materialize/releases

## Upgrade Django helper
- [x] Switch template filters to new set
- [x] Test all the forms manually
- [x] Check that things are working in the areas we customised on the 2016 repo
- [x] Eliminate the dependency and document the change on https://github.com/BirkbeckCTP/django-materialize

### PR DESCRIPTION
**Pull request by joemull**
Related issues:
- Fixes #3199
- Fixes #1134
- Also fixes #3425 if not yet merged, since it's built on that branch

In addition to fixing #3199, this work moves us closer to using class-based views for article list pages.

New things include:
- templates, since the templates need to handle various aspects of FilteredArticlesListView in a particular way.
- date facet field (already had datetime)
- order by dropdown as a reusable template element for each theme
- pagination as a reusable template element for each theme

Visually I moved a few things for clarity, and I hid the search form's order by to avoid confusion, but I tried to keep the interaction design similar. 

Searching still takes you to a separate page. In the future I'd like to merge the list and search results pages so that they form one coherent path for finding the article(s) you want.

Screenshots:
Screenshot from 2023-06-06 13-32-39
Screenshot from 2023-06-06 13-08-52
Screenshot from 2023-06-06 13-31-43

### CODE REVIEW

**Review by mauromsl — APPROVED** [review]
Good one, happy to see old Materialize go as a  bonus (especially djang-materialize)

File: src/themes/material/assets/material.js [hunk 1]
```diff
@@ -7,9 +7,9 @@ function figure_downloads() {
 
 function table_downloads() {
 	var tables = $("div[id^=T]");
-	console.log(tables);
```
Comment (ajrbyers) [inline src/themes/material/assets/material.js:10]: This should probably just be removed.

File: src/themes/material/assets/material.js [hunk 2]
```diff
@@ -7,9 +7,9 @@ function figure_downloads() {
...
+	// console.log(tables);
 	$( tables ).each( function( index, element ){
-		console.log(this);
+		// console.log(this);
```
Comment (ajrbyers) [inline src/themes/material/assets/material.js:12]: This should probably just be removed.

**Review by ajrbyers — APPROVED** [review]
Just two suggestions but looks good.

**Pull request merged**